packages/headless/src/modules/brands/rest/brand.controller.ts

```diff
@@ -0,0 +1,142 @@
+import { ObjectValidationPipe } from '../validators/NestedObjectValidator.pipe';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  Query,
+  Res,
+  HttpStatus,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { ApiResponse } from '@nestjs/swagger';
+
+import { Brand } from 'src/entity/brand';
+import { BrandService } from '../services/index';
+import { GetBrandByIdSuccessResponseDto, GetBrandByIdErrorResponseDto } from 'src/modules/brands/dto/getBrandByIdDto';
+import { CreateBrandRequestDto, CreateBrandSuccessResponseDto, CreateBrandErrorResponseDto } from 'src/modules/brands/dto/createBrandDto';
+import { GetAllBrandsErrorResponseDto, GetAllBrandsSuccessResponseDto } from 'src/modules/brands/dto/getAllBrandsDto';
+import { UpdateBrandRequestdto } from 'src/modules/brands/dto/updateBrandDto';
+import { DeleteBrandErrorResponseDto, DeleteBrandSuccessResponseDto } from 'src/modules/brands/dto/deleteBrandDto';
+import { UpdateBrandErrorResponseDto, UpdateBrandSuccessResponseDto } from './../dto/updateBrandDto';
+
+@Controller('brands')
+export class BrandController {
+  constructor(private brandService: BrandService) {}
+
+  @Get('/')
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'All the brands were fetched successfully',
+    type: GetAllBrandsSuccessResponseDto
+  })
+  @ApiResponse({
+    status: HttpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
why INTERNAL_SERVER_ERROR?

Reply:
get all brands implements no logic or gets no paramter, so if there is any error, what error will it be?
bad request is not an error for a simple get request